src/main/cms/cms_menu_imu.c

```diff
@@ -358,44 +353,227 @@ static const void *cmsx_RateProfileWriteback(displayPort_t *pDisp, const OSD_Ent
-    { "THR EXPO",    OME_UINT8,  NULL, &(OSD_UINT8_t) { &rateProfile.thrExpo8,          0,  100,  1}, 0 },
-    { "THRPID ATT",  OME_FLOAT,  NULL, &(OSD_FLOAT_t) { &rateProfile.dynThrPID,         0,  100,  1, 10}, 0 },
-    { "TPA BRKPT",   OME_UINT16, NULL, &(OSD_UINT16_t){ &rateProfile.tpa_breakpoint, 1000, 2000, 10}, 0 },
+static OSD_Entry cmsx_menuRateProfileEntries[] =
+{
+    { "-- RATE --",    OME_Label, NULL, rateProfileIndexString, 0 },
+    { "RATES TYPE",    OME_TAB,   &cmsx_rateTypeChange, &(OSD_TAB_t) { &rateProfile.rates_type, RATES_TYPE_COUNT - 1, lookupTableRatesType }, 0 },
+
+    // The following entries are dynamically updated based on the rates type
+    { "",              OME_UINT8_SCALED, &cmsx_rateParamChange, &(OSD_UINT8_SCALED_t) { &rateProfile.rcRates[FD_ROLL],  0, 0, 0, 0 }, 0 },
+    { "",              OME_UINT8_SCALED, &cmsx_rateParamChange, &(OSD_UINT8_SCALED_t) { &rateProfile.rates[FD_ROLL],    0, 0, 0, 0 }, 0 },
+    { "",              OME_UINT8_SCALED, &cmsx_rateParamChange, &(OSD_UINT8_SCALED_t) { &rateProfile.rcExpo[FD_ROLL],   0, 0, 0, 0 }, 0 },
+    { "--- MAX DEG/SEC", OME_UINT16, NULL, &(OSD_UINT16_t){ &rcMaxRate[FD_ROLL], 0, 2000, 0 }, DYNAMIC },
 
-    { "THR LIM TYPE",OME_TAB,    NULL, &(OSD_TAB_t)   { &rateProfile.throttle_limit_type, THROTTLE_LIMIT_TYPE_COUNT - 1, osdTableThrottleLimitType}, 0 },
-    { "THR LIM %",   OME_UINT8,  NULL, &(OSD_UINT8_t) { &rateProfile.throttle_limit_percent, 25,  100,  1}, 0 },
+    { "",              OME_UINT8_SCALED, &cmsx_rateParamChange, &(OSD_UINT8_SCALED_t) { &rateProfile.rcRates[FD_PITCH], 0, 0, 0, 0 }, 0 },
+    { "",              OME_UINT8_SCALED, &cmsx_rateParamChange, &(OSD_UINT8_SCALED_t) { &rateProfile.rates[FD_PITCH],   0, 0, 0, 0 }, 0 },
+    { "",              OME_UINT8_SCALED, &cmsx_rateParamChange, &(OSD_UINT8_SCALED_t) { &rateProfile.rcExpo[FD_PITCH],  0, 0, 0, 0 }, 0 },
+    { "--- MAX DEG/SEC", OME_UINT16, NULL, &(OSD_UINT16_t){ &rcMaxRate[FD_PITCH], 0, 2000, 0 }, DYNAMIC },
+
+    { "",              OME_UINT8_SCALED, &cmsx_rateParamChange, &(OSD_UINT8_SCALED_t) { &rateProfile.rcRates[FD_YAW],   0, 0, 0, 0 }, 0 },
+    { "",              OME_UINT8_SCALED, &cmsx_rateParamChange, &(OSD_UINT8_SCALED_t) { &rateProfile.rates[FD_YAW],     0, 0, 0, 0 }, 0 },
+    { "",              OME_UINT8_SCALED, &cmsx_rateParamChange, &(OSD_UINT8_SCALED_t) { &rateProfile.rcExpo[FD_YAW],    0, 0, 0, 0 }, 0 },
+    { "--- MAX DEG/SEC", OME_UINT16, NULL, &(OSD_UINT16_t){ &rcMaxRate[FD_YAW], 0, 2000, 0 }, DYNAMIC },
 
     { "BACK", OME_Back, NULL, NULL, 0 },
     { NULL, OME_END, NULL, NULL, 0 }
 };
 
+
+#define RATE_ENTRY_LABEL_MAX_LENGTH 14   // The maximum length of the longest composite rate param label & axis name. Longest currently is "PITCH RC CURVE"
+
+static char ratesEntriesLabels[9][RATE_ENTRY_LABEL_MAX_LENGTH + 1]; // Stores the constructed menu labels
+
+typedef struct ratesTableEntryType_s {
+    const char * labelText;
+    uint8_t min;
+    uint8_t step;
```

Comment:
Step seems to be 1 for all current cases. Can this be removed until it is needed?

Reply:
I'll test to see if it makes any significant difference.

Reply:
I tried removing the `step` from the `OME_UINT8_SCALED` type and it didn't seem to make any difference in any of the flash blocks. Also the problem with doing this is that the `OME_UINT8_SCALED` structure is shared with the `OME_UINT8` and `OME_FLOAT` structures and they use common increment/decrement code as in:

![Screen Shot 2021-02-14 at 8 51 32 AM](https://user-images.githubusercontent.com/17088539/107878620-f7232e00-6ea1-11eb-8369-1bd415429307.png)

So splitting this apart to handle the `OME_UINT8_SCALED` not having a `step` element seems to offset any potential savings.